src/fragments/lib/datastore/flutter/relational/delete-snippet.mdx

```diff
@@ -1,7 +1,13 @@
 ```dart
-(await Amplify.DataStore.query(Post.classType, where: Post.ID.eq('123')))
-    .forEach((element) async {
-  await Amplify.DataStore.delete(element);
-  print('Deleted a post');
-});
+Future<void> deletePostWithId(String id) async {
+  final postsWithId = await Amplify.DataStore.query(
+    Post.classType,
+    where: Post.ID.eq(id),
```

Comment:
nit: This predicate ensure returns an empty list or a list has only one element, the following for loop in the code example may be unnecessary.